base/strings/util.jl

```diff
@@ -513,6 +513,138 @@ function rpad(
+
+# Examples
+```jldoctest
+julia> s = rtrunc("🍕🍕 I love 🍕", 10)
+"🍕🍕 I lo…"
+
+julia> textwidth(s)
+10
+
+julia> rtrunc("foo", 3)
+"foo"
+```
+
+!!! compat "Julia 1.12"
+    This function was added in Julia 1.12.
+
+See also [`ltrunc`](@ref) and [`ctrunc`](@ref).
+"""
+function rtrunc(str::AbstractString, maxwidth::Int, replacement::Union{AbstractString,Char} = '…')
+    ret = string_truncate_boundaries(str, maxwidth, replacement, Val(:right))
+    if isnothing(ret)
+        return str
+    else
+        left, _ = ret::Tuple{Int,Int}
+        @views return str[begin:left] * replacement
+    end
+end
+
+"""
+    ltrunc(str::AbstractString, maxwidth::Int, replace_str::AbstractString = "…")
+
+Truncate `str` to at most `maxwidth` columns (as estimated by [`textwidth`](@ref)), replacing the first characters
+with `replacement` if necessary. The default replacement string is "…".
+
+# Examples
+```jldoctest
+julia> s = ltrunc("🍕🍕 I love 🍕", 10)
+"…I love 🍕"
+
+julia> textwidth(s)
+10
+
+julia> ltrunc("foo", 3)
+"foo"
+```
+
+!!! compat "Julia 1.12"
+    This function was added in Julia 1.12.
+
+See also [`rtrunc`](@ref) and [`ctrunc`](@ref).
+"""
+function ltrunc(str::AbstractString, maxwidth::Int, replacement::Union{AbstractString,Char} = '…')
+    ret = string_truncate_boundaries(str, maxwidth, replacement, Val(:left))
+    if isnothing(ret)
+        return str
+    else
+        _, right = ret::Tuple{Int,Int}
+        @views return replacement * str[right:end]
+    end
+end
+
+"""
+    ctrunc(str::AbstractString, maxwidth::Int, replacement::Union{AbstractString,Char} = '…'; prefer_left::Bool = true)
+
+Truncate `str` to at most `maxwidth` columns (as estimated by [`textwidth`](@ref)), replacing the middle characters
+with `replacement` if necessary. The default replacement string is "…". By default, the truncation
+prefers keeping chars on the left, but this can be changed by setting `prefer_left` to `false`.
+
+# Examples
+```jldoctest
+julia> s = ctrunc("🍕🍕 I love 🍕", 10)
+"🍕🍕 …e 🍕"
+
+julia> textwidth(s)
+10
+
+julia> ctrunc("foo", 3)
+"foo"
+```
+
+!!! compat "Julia 1.12"
+    This function was added in Julia 1.12.
+
+See also [`ltrunc`](@ref) and [`rtrunc`](@ref).
+"""
+function ctrunc(str::AbstractString, maxwidth::Int, replacement::Union{AbstractString,Char} = '…'; prefer_left::Bool = true)
+    ret = string_truncate_boundaries(str, maxwidth, replacement, Val(:center), prefer_left)
+    if isnothing(ret)
+        return str
+    else
+        left, right = ret::Tuple{Int,Int}
+        @views return str[begin:left] * replacement * str[right:end]
+    end
+end
+
+function string_truncate_boundaries(
+            str::AbstractString,
+            maxwidth::Int,
+            replacement::Union{AbstractString,Char},
+            ::Val{mode},
+            prefer_left::Bool = true) where {mode}
+
+    maxwidth >= 0 || throw(ArgumentError("maxwidth $maxwidth should be non-negative"))
```

Comment:
```suggestion
    width = textwidth(replacement)
    maxwidth >= width || throw(ArgumentError("maxwidth $maxwidth should be ≥ $(width), the width of $(repr(replacement))"))
```
and delete the `width = textwidth(replacement)` line below.

Reply:
@tecosaur and I talked about this previously and concluded that it would be better for terminal usage to not error when `replacement` is wider than `maxwidth`, because for instance if the terminal window is resized down such that that's true, it would be better to just print `replacement` (or the first `maxwidth` chars of `replacement`)  rather than throw.